rbtcwrapperproxy/contracts/RBTCWrapperProxy.sol

```diff
@@ -374,24 +358,22 @@ contract RBTCWrapperProxy is ContractRegistryClient {
 
         uint256 reserveAmount;
         bool successOfTransfer;
-        for (uint256 i = 1; i < lengthOfToken; i++) {
-
-            reserveAmount = _reserveTokens[i].balanceOf(address(this)).sub(reserveAmountBefore[i]); 
-            require(reserveAmount >= _reserveMinReturnAmounts[i], "min return too high");
-            reserveAmounts[i] = reserveAmount;
-            
-
-            successOfTransfer = IERC20Token(_reserveTokens[i]).transfer(msg.sender, reserveAmount);
-            require(successOfTransfer, "Failed to transfer reserve token to user");
+        for (uint256 i = 0; i < lengthOfToken; i++) {
+            if(address(_reserveTokens[i]) == wrbtcTokenAddress) {
+                uint256 wrbtcAmount = _reserveTokens[i].balanceOf(address(this)).sub(reserveAmountBefore[i]);
+                reserveAmounts[i] = wrbtcAmount;
+                IWrbtcERC20(wrbtcTokenAddress).withdraw(wrbtcAmount);
+                (bool successOfSendRBTC,) = msg.sender.call.value(wrbtcAmount)("");
+                require(successOfSendRBTC, "Failed to send RBTC to user");
+            } else {
+                reserveAmount = _reserveTokens[i].balanceOf(address(this)).sub(reserveAmountBefore[i]); 
+                reserveAmounts[i] = reserveAmount;
+
+                successOfTransfer = IERC20Token(_reserveTokens[i]).transfer(msg.sender, reserveAmount);
+                require(successOfTransfer, "Failed to transfer reserve token to user");
+            }
```

Comment:
FN3 - can in be aggregated with _refund function?

Reply:
We need to handle different behavior of reserveAmounts value for event purposes. So i don't think aggregating is good approach here